### PR TITLE
fromString util method

### LINKED ISSUE #259: Add "fromString" factory constructor for PowerAuthPassword object
### Task specification

In the same manner as in the flutter: https://developers.wultra.com/components/flutter-powerauth-mobile-sdk/1.1.x/documentation/Secure-Password#using-password

### PR DESCRIPTION
**Pull request by kober32**
Fixes #259

### CODE REVIEW

File: testapp/tsconfig.json [hunk 1]
```diff
@@ -1,9 +1,10 @@
 {
   "extends": "@react-native/typescript-config/tsconfig.json",
   "compilerOptions": {
+    "types": ["react-native"],
     "strict": true,
-    "noUncheckedIndexedAccess": true,
-    "noImplicitOverride": true,
+    // "noUncheckedIndexedAccess": true,
```
Comment (kober32) [inline testapp/tsconfig.json:6]: I had to disable those 2 rules, because otherwise the test app produces a lot of errors, when we don't use `override` or are using unchecked index access...

I think this is OK for a test app...

File: docs/Secure-Password.md [hunk 2]
```diff
@@ -61,6 +61,16 @@ try {
 }
 ```
 
+You can also create `PowerAuthPassword` from an already obtained string.
+
+<!-- begin box warning -->
+Note that this is not recommended. Do this only when you retrieve the whole string from a text input.
```
Comment (marek-ch) [inline docs/Secure-Password.md:67]: I do not fully understand the warning. Why it is not recommended? Is it ment to be called one time for full text input? Instead of calling it every time user inputs another char?

File: packages/react-native-powerauth-mobile-sdk/src/model/PowerAuthPassword.ts [hunk 3]
```diff
@@ -88,6 +88,29 @@ export class PowerAuthPassword extends BaseNativeObject {
...
+     * Creating password from already obtained string.
+     * 
+     * Note that this is not recommended. Do this only when you retrieve the whole string from a text input.
+     * 
+     * @param destroyOnUse If `true` then the native password is destroyed after is used for the cryptographic operation. Default is `true`.
+     * @param onAutomaticCleanup If provided, then the closure is called when the native password is restored and the previous content is lost.
+     * @param powerAuthInstanceId If specified, then the native password will be destroyed together with PowerAuth instance.
+     * @param autoreleaseTime Autorelease timeout in milliseconds. The value is used only for the testing purposes, and is ignored in the release build of library.
+     */
+    static async fromString(
```
Comment (marek-ch) [inline packages/react-native-powerauth-mobile-sdk/src/model/PowerAuthPassword.ts:101]: just thing for thought (as I do not have as much context around), but I feel like this `fromString` is much more like `addCharacters`?  Because from the usage (in tests) I see that we have `new PowerAuthPassword()` as the starting point with empty password.
So my suggestions here would be:
- create constuctor `PowerAuthPassword(from: string)` - static method seems odd here.
- or rename it to `addCharacters(chars: string)` that are more in-line with existing `addCharacter` implementation
This is my external opinions :)

**Review by marek-ch — DISMISSED** [review]
code is ok, comments are optional :)

File: packages/react-native-powerauth-mobile-sdk/src/model/PowerAuthPassword.ts (hunk 3 above)
Comment (marek-ch) [reply packages/react-native-powerauth-mobile-sdk/src/model/PowerAuthPassword.ts:101]: but there is no async constructor in JS. 
So in that case I would go with `addCharacters`, leaving the rest of parameters to existing constructor

File: docs/Secure-Password.md (hunk 2 above)
Comment (kober32) [reply docs/Secure-Password.md:67]: Yes, exactly, we don't want the integrator to store PIN to temp. string field, because it creates a memory footprint. So when implementing the pin keyboard, addcharacter method should be used. So this is meant to be used only for text/password fields.

File: packages/react-native-powerauth-mobile-sdk/src/model/PowerAuthPassword.ts (hunk 3 above)
Comment (kober32) [reply packages/react-native-powerauth-mobile-sdk/src/model/PowerAuthPassword.ts:101]: Hmm, good point with the naming, we have it already in flutter, but I will discuss it with Martin and we'll see...

**Review by mbudinsky — DISMISSED** [review]

**Review by mbudinsky — APPROVED** [review]
